### PR TITLE
Rooms

### PR DESCRIPTION
**Pull request by danielaKokoneshi309**
Room CRUD operations

### CODE REVIEW

**Review by Akeron44 — APPROVED** [review]

File: middleware/auth.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const jwt = require('jsonwebtoken');
+
+
+module.exports = function (req, res, next) {
+  const token = req.header('x-auth-token');
+  if (!token) return res.status(401).send('Access denied. No token provided.');
```
Comment (KostandinD) [inline middleware/auth.js]: Hey, I think a better message would be something like: **Token is missing or invalid. Please provide a valid authentication token.**

File: routes/room.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+const validateRoom = require('../validation/roomValidation');
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+    try {
+        const room = await Room.find({});
+        res.send(room);
+    } catch (err) {
+        console.log(err);
```
Comment (KostandinD) [inline routes/room.js]: Please, always make sure to remove `console.log()` before you create the PR.

File: routes/room.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+const express = require('express');
+const {Room} = require('../models/room');
```
Comment (KostandinD) [inline routes/room.js]: It would be nice if here you cloud add some spacing and transform `{Room} ` to `{ Room }`.

File: routes/room.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+router.get('/', async (req, res) => {
+    try {
+        const room = await Room.find({});
+        res.send(room);
+    } catch (err) {
+        console.log(err);
+        res.status(500).send('Error retrieving data');
+    }
+});
+router.get('/:id', async (req, res) => {
```
Comment (KostandinD) [inline routes/room.js]: Please, make sure to add line spacing when it makes sense.

File: routes/room.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    try {
+        const room = await Room.findById(req.params.id);
+        if (!room) return res.status(404).send('The room was not found');
+        res.send(room);
+    } catch (err) {
+        res.status(400).send('Invalid ID format');
```
Comment (KostandinD) [inline routes/room.js]: Hey, keep in mind that the error that should be thrown here not necessarily would have to do with **Invalid ID format**. What I would suggest is to have another check prior to the line which finds the object, where you check if the id is valid or not and throw the respective error there.

File: routes/room.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        });
+
+        const savedRooms = await room.save();
+        
+        res.send(savedRooms);
+    } catch (err) {
+        res.status(500).send('An error occurred while saving the rooms.');
+    }
+});
+router.put('/:id', async (req, res) => {
```
Comment (KostandinD) [inline routes/room.js]: Please, in this code block make sure the spacing is applied. Also here I would suggest to add the provided id validation.

File: startup/routes.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 
 const express = require("express");
-
+const rooms= require('../routes/room');
```
Comment (KostandinD) [inline startup/routes.js]: Please, add a space before the `=` sign.

File: startup/routes.js [hunk 8]
```diff
@@ -9,5 +9,5 @@ require("express-async-errors");
 module.exports = function (app) {
   app.use(express.json());
  
-
+  app.use('/api/room',rooms);
```
Comment (KostandinD) [inline startup/routes.js]: In the route naming, not necessarily you need to add `/api`. However, make sure to discuss it with your teammates and decide on your preferred approach.

File: validation/roomValidation.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+const Joi = require('joi');
```
Comment (KostandinD) [inline validation/roomValidation.js]: Please, also here be mindful of spacing and how you add it. Also, please follow the structure we discussed and add the validation inside a folder together with the route.

**Review by KostandinD — CHANGES_REQUESTED** [review]
The logical part and the overall code is OK. However, please be more consistent with clean code principals.

**Review by KostandinD — APPROVED** [review]
Please, make sure to align accordingly with the rest of the teammates, so the unnecessary extra code is remove.
Also, the message of the commit should be descriptive. In this case, I would have suggested the message: "Code refactoring".

**Pull request merged**